core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -232,26 +238,17 @@ public X509Certificate getTrustedCertificateAuthority() throws CertificateExcept
      * Once a (non {@code null}) trust anchor has been created, it will be cached and
      * returned on subsequent invocations of this method.
      * 
+     * @deprecated Replaced by {@link #getTrustAnchors()}
      * @return The trust anchor or {@code null} if no trusted certificate authority
      *         has been set.
      * @throws GeneralSecurityException if the value of the <em>trusted-ca</em> property
-     *             cannot be parsed into a trust anchor, e.g. because of unsupported
-     *             key type, malformed certificate or public key encoding etc. 
+     *              cannot be parsed into a trust anchor, e.g. because of unsupported
+     *              key type, malformed certificate or public key encoding etc.
      */
+    @Deprecated(since="1.0", forRemoval=true)
```

Comment:
I do not think that we should deprecate this method. Instead, we should simply replace it with *getTrustAnchors*.

Reply:
if i should have replaced it in this PR, then i will have to change the *-adapters code base that uses it when authenticating client devices using x.509 certificates. i decided to do that in a separate PR (and to also add some integration tests for it). This is the reason why i deprecated it in this PR.